frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -188,6 +219,94 @@ function MainMenu(props: Props): ReactElement {
   const { colors }: Theme = useTheme()
   const isServerDisconnected = !props.isServerConnected
 
+  const onClickDeployApp = useCallback((): void => {
+    const {
+      showDeployError,
+      closeDialog,
+      isDeployErrorModalOpen,
+      gitInfo,
+    } = props
+
+    if (!gitInfo) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    const {
+      repository,
+      branch,
+      module,
+      untrackedFiles,
+      uncommittedFiles,
+      state: gitState,
+    } = gitInfo
+
+    if (
+      (!repository || !branch || !module) &&
+      gitState === GitStates.DEFAULT
+    ) {
+      const dialog = NoRepositoryDetected()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (gitState === GitStates.HEAD_DETACHED) {
+      const dialog = DetachedHead()
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (module && untrackedFiles?.includes(module)) {
+      const dialog = ModuleIsNotAdded(module)
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (repository && uncommittedFiles?.length) {
+      const dialog = UncommittedChanges(repository)
+
+      showDeployError(dialog.title, dialog.body)
+
+      return
+    }
+
+    if (gitState === GitStates.AHEAD_OF_REMOTE) {
+      const dialog = RepoIsAhead()
+
+      showDeployError(dialog.title, dialog.body, getDeployAppUrl(gitInfo))
+
+      return
+    }
+
+    if (untrackedFiles?.length) {
+      const dialog = UntrackedFiles()
+
+      showDeployError(dialog.title, dialog.body, getDeployAppUrl(gitInfo))
+
+      return
+    }
+
+    // We should close the modal when we try again and everything goes fine
+    if (isDeployErrorModalOpen) closeDialog()
```

Comment:
Nit: I believe we agreed at some point not to use single-line if statements. The reason being that we wanted our code to look consistent. We should probably make this an ESLint rule some day.